Index: rtl/freertos/xtensa/esp8266.pp
===================================================================
--- rtl/freertos/xtensa/esp8266.pp	(revision 47785)
+++ rtl/freertos/xtensa/esp8266.pp	(working copy)
@@ -81,7 +81,11 @@
       begin
         ReadChar := true;
         ACh := #0;
-        uart_rx_one_char(@ACh);  // check failure?
+        repeat
+          uart_rx_one_char(@ACh);  // check failure?
+          if ACh = #0 then
+            vTaskDelay(1);
+        until ACh <> #0;
       end;
 
 begin
